core/services/telemetry/manager.go

```diff
@@ -0,0 +1,172 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type Client interface {
+	services.ServiceCtx
+	Send(payload synchronization.TelemPayload)
```

Comment:
No ctx as first parameter?

Reply:
It get's pushed to a buffer and later sent, that's why we don't have a context param

Reply:
+1 to ctx. @george-dorin it sounds like you already know implementation details to omit the context. i'd rather see the interface be fully formed with a ctx and hide the implementation details

Reply:
Updated to include context in https://github.com/smartcontractkit/chainlink/pull/10623/commits/47032d4993460ca3d812155241b496a06ab5166d